draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+The following keywords are roughly analogous to web links:
+
+* The `"$id"` keyword effectively defines a link with relation type "self".
+* Reference keywords with object semantics effectively defines a link with relation type "full".
+* Reference keywords with keyword semantics effectively defines a link with the generic relation type "related".
+
+In the absence of other link metadata or context specification guidance, the target of a reference keyword MAY be treated as carrying a target media type hint (as in {{?RFC8288, Section 3.4.1}}) suggesting that the target media type is identical to the media type of the document containing the reference.
+
+# Interoperability Considerations {#interop}
+
+See also {{interop-subset}}.
+
+## JRI behavior negotiation
+
+To maximize the usefulness of standalone tooling, such tools need to be able to determine the degree to which JRI is used and the interoperability assurances on which it can rely.[^41]
+
+[^41]: This section is more-or-less brainstorming.  The problem it is solving is real, but whether these solutions belong in this specification or not is open to question.
+
+### Media type parameters for safe processing
+
+Context specifications defining media types SHOULD define the `jri-safety` and `jri-keywords` media type parameters to enable standalone implementations, including standalone subset implementations, to determine whether they can safely process the document.
+
+These parameters are intended for use in environments where a standalone implementation is likely to encounter documents with unexpected media types involving unpredictable JRI usage.  Standalone implementations MAY use other heuristics and information sources to determine whether to process a document.
+
+Non-media type context data formats MAY define a mechanism analogous to these media type parameters, but such mechanisms will not be interoperable.
+
+#### The "jri-safety" media type parameter
+
+The value of the `jri-safety` parameter MUST be case-insensitive, and MUST be either `interoperable` or `contextual`.  The default value is `contextual`, indicating that an implementation MUST be aware of and support the relevant context specification in order to safely process JRI keywords in the document.  A value of `interoperable` indicates that the document is context-independent, and safe to process by a standalone implementation.
+
+### The "jri-keywords" media type parameter
+
+The value of the `jri-keywords` parameter MUST be a case-sensitive comma-separated list of JRI keyword names, which MUST NOT contain whitespace.  These keywords MAY be in any order.  This indicates that only the listed keywords are used in the document. In the absence of this parameter, an implementation MUST assume that any JRI keyword can be used.
```

Comment:
the items SHOULD be unique (i.e. no duplicates)?
or define this as a "set" instead of a "list"?